Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/LoadTestDataDBServiceImpl.java

```diff
@@ -46,66 +52,157 @@
+    public List<Identifier> getAllIdentiersBy(Long patientId) {
+        return identifierDAO.findRecords(patientId);
+    }
+
+    @Override
+    public List<Phonenumber> getAllPhonenumbersBy(Long patientId) {
+        return phonenumberDAO.findRecords(patientId);
+    }
+
+    @Override
+    public boolean deletePersonname(Personname personname) {
+        return personnameDAO.delete(personname);
+    }
+
+    @Override
+    public boolean deleteAddress(Address address) {
+        return addressDAO.delete(address);
+    }
+
+    @Override
+    public boolean deleteIdentifier(Identifier identifier) {
+        return identifierDAO.delete(identifier);
+    }
+
+    @Override
+    public boolean deletePhonenumber(Phonenumber phonenumber) {
+        return phonenumberDAO.delete(phonenumber);
+    }
 
+    @Override
+    public Personname getPersonnameBy(Long personnameId) {
+        return personnameDAO.read(personnameId);
+    }
+
+    @Override
+    public Address getAddressBy(Long addressId) {
+        return addressDAO.read(addressId);
+    }
+
+    @Override
+    public Identifier getIdentifierBy(Long identifierId) {
+        return identifierDAO.read(identifierId);
+    }
+
+    @Override
+    public Phonenumber getPhonenumberBy(Long phonenumberId) {
+        return phonenumberDAO.read(phonenumberId);
+    }
+
+    @Override
+    public boolean savePersonname(Personname personname) throws LoadTestDataException {
+        boolean actionResult = false;
+        if (personname.getPatient() != null) {
+            actionResult = personnameDAO.save(personname);
             if (!actionResult) {
-                throw new LoadTestDataException("Patient cannot be save.");
+                logPatientError("Personname");
             }
+        }
+        return actionResult;
+    }
 
+    @Override
+    public boolean saveAddress(Address address) throws LoadTestDataException {
+        boolean actionResult = false;
+        if (address.getPatient() != null) {
+            actionResult = addressDAO.save(address);
+            if (!actionResult) {
+                logPatientError("Address");
+            }
         }
-        else {
-            throw new LoadTestDataException(
-                "LoadTestDataDBServiceImpl Patient-Personname is required when trying to save-patient.");
+        return actionResult;
+    }
+
+    @Override
+    public boolean saveIdentifier(Identifier identifier) throws LoadTestDataException {
+        boolean actionResult = false;
+        if (identifier.getPatient() != null) {
+            actionResult = identifierDAO.save(identifier);
+            if (!actionResult) {
+                logPatientError("Identifier");
+            }
         }
+        return actionResult;
+    }
 
-        LOG.info("LoadTestDataDBServiceImpl save-patient: Action-result: {} ", actionResult);
+    @Override
+    public boolean savePhonenumber(Phonenumber phonenumber) throws LoadTestDataException {
+        boolean actionResult = false;
```

Comment:
Handling boolean - Be consistent with GenericDao or change the default value check in GeneriDao

Reply:
fixed in the dao